### DISCUSSION TITLE
Inter-process compatibility?

### DISCUSSION THREAD

**Discussion by kikaragyozov** [samuel-lucas6/Geralt#4] [Q&A] [answered]
Does the library do some magic that would break in the event of inter-process communication between different clients/companies whom use their own libraries/implementations for encryption/decryption?

The concrete case I'm wondering specifically about, is signing data with your `Ed25519` implementation.

**Comment by samuel-lucas6** [comment, marked as answer]
Most functions, including everything in the `Ed25519` class, call libsodium directly without doing anything special. Therefore, such functions should be compatible with other libsodium wrappers and correct implementations of the algorithms.

The thing about Ed25519 in libsodium though is that the `privateKey` actually includes the `publicKey`. This saves computation and helps [prevent](https://github.com/MystenLabs/ed25519-unsafe-libs) private key exposure from using an incorrect public key. Other libraries handle this issue [differently or not at all](https://github.com/LoupVaillant/Monocypher/issues/240), although this is quite a common technique.

Hope that helps. Let me know if that didn't answer your question.

**Reply by kikaragyozov** [reply]
> Most functions, including everything in the `Ed25519` class, call libsodium directly without doing anything special. Therefore, such functions should be compatible with other libsodium wrappers and correct implementations of the algorithms.
> 
> The thing about Ed25519 in libsodium though is that the `privateKey` actually includes the `publicKey`. This saves computation and helps [prevent](https://github.com/MystenLabs/ed25519-unsafe-libs) private key exposure from using an incorrect public key. Other libraries handle this issue [differently or not at all](https://github.com/LoupVaillant/Monocypher/issues/240), although this is quite a common technique.
> 
> Hope that helps. Let me know if that didn't answer your question.

Thank you for the descriptive answer! This is out of the scope of the question, but are you familiar with any good libraries that support generating OpenPGP ECDSA/EdDSA keys?

**Reply by samuel-lucas6** [reply]
I expect [Bouncy Castle](https://bouncycastle.org/csharp/index.html) supports that, although understanding their API will probably take some trial and error.
